flake.nix

```diff
@@ -132,6 +138,8 @@
           export RUST_BACKTRACE=1
           export ZSTD_SYS_USE_PKG_CONFIG=1
           export LIBSSH2_SYS_USE_PKG_CONFIG=1
+          export NIX_JJ_GIT_HASH="${gitRev}"
+          export NIX_JJ_GIT_TIMESTAMP="${gitTimestamp}";
```

Comment:
So this will ensure `NIX_JJ_GIT_TIMESTAMP` (and `NIX_JJ_GIT_HASH`) only actually gets configured when the environment is populated, which typically happens only when you run `nix develop` for the first time. I think this is probably OK, to be honest, since it's only a minor glitch when nix-oriented developers are hacking. It just means the `--version` will be wrong, but that's probably OK in the working copy.

To fix this we'd need to have a third case where we set something like `NIX_JJ_SHELL=1` and then have `build.rs` handle that case like the normal one where we just interrogate the `.git` dir.

Reply:
Ideally, these variables would only populate in the case when jj is compelled outside a git repo by nix.

It's beyond the scope of my comfort level with nix, though. Somebody else would have to test these things and make another PR, I think.